drivers/clock_control/clock_control_nrf2_hfxo.c

```diff
@@ -48,6 +50,41 @@ static void hfxo_start_up_timer_handler(struct k_timer *timer)
 	}
 }
 
+static void start_hfxo(const struct device *dev)
+{
+	struct dev_data_hfxo *dev_data = dev->data;
+
+	nrf_lrcconf_event_clear(NRF_LRCCONF010, NRF_LRCCONF_EVENT_HFXOSTARTED);
+	soc_lrcconf_poweron_request(&dev_data->hfxo_node, NRF_LRCCONF_POWER_MAIN);
+	nrf_lrcconf_task_trigger(NRF_LRCCONF010, NRF_LRCCONF_TASK_REQHFXO);
+}
+
+static void request_hfxo(const struct device *dev)
+{
+	struct dev_data_hfxo *dev_data = dev->data;
+	unsigned int key;
+
+	key = irq_lock();
+	if (!dev_data->requested && dev_data->req_count_zli != 0) {
```

Comment:
Why `dev_data->req_count_zli != 0`?

Reply:
This is error. It looks it didn't make a problem because in my tests BLE/ZLI context was only one user of HFXO.